client/blocks/upload-image/index.jsx

```diff
@@ -0,0 +1,174 @@
+	};
+
+	static defaultProps = {
+		imageEditorProps: {
+			allowedAspectRatios: [ AspectRatios.ASPECT_1X1 ],
+		},
+		texts: {},
+		backgroundContent: null,
+		onImageEditorDone: noop,
+		isUploading: false,
+	};
+
+	receiveFiles = files => {
+		const extension = path.extname( files[ 0 ].name ).toLowerCase().substring( 1 );
+
+		if ( ALLOWED_FILE_EXTENSIONS.indexOf( extension ) === -1 ) {
+			return;
+		}
+
+		const imageObjectUrl = URL.createObjectURL( files[ 0 ] );
+
+		this.setState( {
+			isEditingImage: true,
+			uploadedImage: imageObjectUrl,
+		} );
+	};
+
+	onImageEditorDone = ( error, imageBlob ) => {
+		this.hideImageEditor();
+
+		if ( error ) {
+			return;
+		}
+
+		this.setState( { editedImage: URL.createObjectURL( imageBlob ) } );
+
+		this.props.onImageEditorDone( imageBlob );
+	};
+
+	hideImageEditor = () => {
+		const { resetAllImageEditorState: resetAllImageEditorStateAction } = this.props;
+
+		resetAllImageEditorStateAction();
+
+		URL.revokeObjectURL( this.state.uploadedImage );
+
+		this.setState( {
+			isEditingImage: false,
+			uploadedImage: false,
+		} );
+	};
+
+	renderImageEditor() {
+		const { isEditingImage, uploadedImage } = this.state;
+
+		if ( ! isEditingImage ) {
+			return null;
+		}
+
+		const { additionalImageEditorClasses, imageEditorProps, texts } = this.props;
+
+		const classes = classnames( 'upload-image-modal', additionalImageEditorClasses );
+
+		return (
+			<Dialog additionalClassNames={ classes } isVisible={ true }>
+				<ImageEditor
+					{ ...imageEditorProps }
+					media={ { src: uploadedImage } }
+					onDone={ this.onImageEditorDone }
+					onCancel={ this.hideImageEditor }
+					doneButtonText={ texts.doneButtonText ? texts.doneButtonText : 'Done' }
+				/>
+			</Dialog>
+		);
+	}
+
+	render() {
+		const { children, isUploading, translate, texts, additionalClasses } = this.props;
+
+		let { placeholderContent, uploadingContent } = this.props;
+
+		if ( typeof placeholderContent === 'undefined' ) {
+			placeholderContent = (
+				<div className="upload-image__placeholder">
+					<Gridicon icon="add-image" size={ 36 } />
+					<span>
+						{ texts.addAnImage ? texts.addAnImage : translate( 'Add an Image' ) }
+					</span>
+				</div>
+			);
+		}
+
+		if ( typeof uploadingContent === 'undefined' ) {
+			const { editedImage } = this.state;
+
+			uploadingContent = (
+				<div className="upload-image__uploading-container">
+					<img src={ editedImage } />
+					<Spinner className="upload-image__spinner" size={ 40 } />
+				</div>
+			);
+		}
+
+		return (
+			<div className={ classnames( 'upload-image', additionalClasses ) }>
+				{ this.renderImageEditor() }
+
+				<FilePicker accept="image/*" onPick={ this.receiveFiles }>
+					<div
+						className={ classnames( 'upload-image__image-container', {
+							'is-uploading': isUploading,
+						} ) }
+					>
+						<DropZone
+							textLabel={
+								texts.dragUploadText ? texts.dragUploadText : translate( 'Drop to upload image' )
```

Comment:
Hi! I've found a possible matching string that has already been translated **29** times:
`translate( 'Drop images to upload' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).